akka-http-core/src/main/scala/akka/http/scaladsl/model/Uri.scala

```diff
@@ -215,6 +215,12 @@ object Uri {
   def apply(input: ParserInput, charset: Charset, mode: Uri.ParsingMode): Uri =
     new UriParser(input, charset, mode).parseUriReference()
 
+  def apply(uri: java.net.URI): Uri = {
```

Comment:
TODO, should we provide this method that'd take the various URI parts from the uri and pass them in directly rather than going via strings and parsing? Sounds like a good idea I thought

Reply:
We could do this but the question is if it is any better (better performing, correcter) than going through the well specified URI string representation.